src/app/tabs/newAdmin/addresses/operations.js

```diff
@@ -103,6 +122,57 @@ const setMultiChainAddress = (domain, chainId, address, isNew) => async (dispatc
   );
 };
 
+/**
+ * Sets values in the Definitive Resolver.
+ * @param {string} domain the domain to be set
+ * @param {uint8} chainId chain Id for the coin
+ * @param {address} address address to be set
+ * @param {bool} isNew is this a new address?
+ */
+const setDefinitiveAddress = (domain, chainId, address, isNew) => async (dispatch) => {
+  const chainName = getChainNameById(chainId);
+  const chainIndex = getIndexById(chainId);
+  const accounts = await window.ethereum.enable();
+  const currentAddress = accounts[0];
+
+  dispatch(requestSetChainAddress(chainName));
+
+  // encode value if it is not empty:
+  let encodeValue = address;
+  if (encodeValue !== EMPTY_ADDRESS) {
+    try {
+      encodeValue = formatsByCoinType[chainIndex].decoder(address);
+    } catch (error) {
+      return dispatch(errorSetChainAddress(chainName, ADDRESS_ENCODING_ERROR, address));
+    }
+  }
+
+  return definitiveResolver.methods.setAddr(namehash(domain), chainId, encodeValue)
+    .send({ from: currentAddress }, (error, result) => {
+      dispatch(waitingSetChainAddress(chainName));
+      if (error) {
+        return dispatch(errorSetChainAddress(chainName, error.message));
+      }
+
+      const transactionConfirmed = () => () => {
+        dispatch(receiveSetChainAddress(
+          chainId, getChainNameById(chainId), address, result, isNew,
+        ));
+        // if deleting, close the error message programatically
+        if (address === '' || address === EMPTY_ADDRESS) {
+          dispatch(closeSetChainAddress(chainName));
+          sendBrowserNotification(domain, 'coin_address_removed');
+        } else {
+          sendBrowserNotification(
+            domain,
+            isNew ? 'coin_address_added' : 'coin_address_updated',
+          );
+        }
+      };
+
+      return dispatch(transactionListener(result, () => transactionConfirmed()));
```

Comment:
This change combined with the one above should work fine. The execution here is not necessary, the parameter is a function

```suggestion
      return dispatch(transactionListener(result, () => transactionConfirmed));
```

Reply:
Thanks, this works. I have this setup in other places in the manager that should be corrected. I added it to my list of desired refactorings.